cmd/relay-metrics/main.go

```diff
@@ -0,0 +1,275 @@
+		Seconds: time.Now().Unix(),
+	}
+	return &monitoringpb.TimeSeries{
+		Metric: &metricpb.Metric{
+			Type: metricType,
+			Labels: map[string]string{
+				"environment": environment,
+			},
+		},
+		Resource: &monitoredres.MonitoredResource{
+			Type: "global",
+			Labels: map[string]string{
+				"project_id": projectID,
+			},
+		},
+		Points: []*monitoringpb.Point{{
+			Interval: &monitoringpb.TimeInterval{
+				EndTime: now,
+			},
+			Value: &monitoringpb.TypedValue{
+				Value: &monitoringpb.TypedValue_Int64Value{
+					Int64Value: int64(value),
+				},
+			},
+		}},
+	}
+}
+
+func writeTimeSeriesRequest(ctx context.Context, mc *monitoring.MetricClient, series []*monitoringpb.TimeSeries) error {
+	req := &monitoringpb.CreateTimeSeriesRequest{
+		Name:       "projects/" + projectID,
+		TimeSeries: series,
+	}
+	log.Printf("writeTimeseriesRequest: %+v\n", req)
+
+	err := mc.CreateTimeSeries(ctx, req)
+	if err != nil {
+		return fmt.Errorf("could not write time series value, %v ", err)
+	}
+	return nil
+}
+
+type workflowRunMetric struct {
+	Name              string
+	Status            string
+	SecondsSinceStart float64
+}
+
+func getStatuses(ctx context.Context, c client.Client) (ret []*workflowRunMetric) {
+	now := time.Now().UTC()
+	wrs := &nebulav1.WorkflowRunList{}
+	err := c.List(ctx, wrs)
+	if err != nil {
+		panic(err.Error())
+	}
+	for _, item := range wrs.Items {
+		m := workflowRunMetric{
+			Name:              item.Name,
+			Status:            item.Status.Status,
+			SecondsSinceStart: now.Sub(item.ObjectMeta.CreationTimestamp.UTC()).Seconds(),
+		}
+		fmt.Printf("%s is %s, %d\n", m.Name, m.Status, int(m.SecondsSinceStart))
+		ret = append(ret, &m)
+	}
+	return
+}
+
+func homeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
+	}
+	return os.Getenv("USERPROFILE") // windows
+}
+
+func k8sClient(kubeconfig *string) client.Client {
+	// Try homedir config
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		// Try in-cluster config
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			panic(err.Error())
+		}
+	}
+
+	c, err := client.New(config, client.Options{
+		Scheme: Scheme,
+	})
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return c
+}
+
+func metricsClient(ctx context.Context) *monitoring.MetricClient {
+	mc, err := monitoring.NewMetricClient(ctx)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return mc
+}
+
+func main() {
+	ctx := context.Background()
+	var err error
+	var kubeconfig *string
+
+	if home := homeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	publish := flag.String("publish", "false", "Publish metrics to stackdriver")
+	environment := flag.String("environment", "testing", "The environment being monitored for workflow runs")
```

Comment:
We use `dev`/`stage`/`prod` as our environment labels.

Reply:
Yeah, it's literally `testing` because I wanted a default that didn't conflict with our existing label usage.

Reply:
I'd defer to use `dev` at this case.